tests/unit/test_loaders.py

```diff
@@ -206,9 +294,116 @@ def test_load_service_model_uses_provided_type_name(self):
     def test_create_loader_parses_data_path(self):
         search_path = os.pathsep.join(['foo', 'bar', 'baz'])
         loader = create_loader(search_path)
-        self.assertIn('foo', loader.search_paths)
-        self.assertIn('bar', loader.search_paths)
-        self.assertIn('baz', loader.search_paths)
+        self.assertIn(pathlib.Path('foo'), loader.search_paths)
+        self.assertIn(pathlib.Path('bar'), loader.search_paths)
+        self.assertIn(pathlib.Path('baz'), loader.search_paths)
+
+    @pytest.mark.skipif(
+        sys.version_info >= (3, 9), reason="Python version >= 3.9"
+    )
+    def test_zips_not_suppored_python_lt_39(self):
+        search_path = os.path.join(
+            os.path.dirname(__file__), 'data', 'Archive.zip', 'foo'
+        )
+        with self.assertRaises(RuntimeError):
+            create_loader(search_path)
+
+    @pytest.mark.skipif(
+        sys.version_info < (3, 9), reason="Python version < 3.9"
+    )
+    def test_zip_path_created(self):
+        search_path = os.path.join(
+            os.path.dirname(__file__), 'data', 'Archive.zip', 'foo'
+        )
+        loader = create_loader(search_path)
+        self.assertIn(
+            search_path + os.sep, [str(path) for path in loader.search_paths]
+        )
+        zip_path = ZipPath(os.path.dirname(search_path)).joinpath('foo')
+        # two identical ZipPath objects do not 'equal' each other
+        self.assertIn(
+            str(zip_path), (str(path) for path in loader.search_paths)
+        )
+        # sanity check that one of the paths is a ZipPath
+        # since it can't be checked directly
+        self.assertTrue(
+            any(isinstance(path, ZipPath) for path in loader.search_paths)
+        )
```

Comment:
This may be a simpler approach and more exact in what we're checking.
```suggestion
        # two identical ZipPath objects do not 'equal' each other
       matching_zips = [path for path in loader.search_paths if str(path) == str(zip_path)]
       assert len(matching_zips) == 1
       assert isinstance(matching_zips[0], ZipPath)
```